### PR TITLE
Always regenerate files wxs fragment

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

* Always regenerate the files wxs fragment
  * Add our own JSON BOM and verify that
    * The goal would be for this BOM to change less often and have more information about our files.  

## PR Context

We no long have using `msp` in scope

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -5062,3 +4765,207 @@ function ConvertTo-PEOperatingSystem {
...
+        [string]
+        $BomName,
+        [ValidateScript({ Test-Path $_ })]
+        [string]
+        $Path,
+        [switch]
+        $Fix
+    )
+
+    $root = (Resolve-Path $Path).ProviderPath -replace "\$([System.io.path]::DirectorySeparatorChar)$"
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:4833]:
```suggestion
    Write-Log "verifying no unauthorized files have been added or removed..."
    $root = (Resolve-Path $Path).ProviderPath -replace "\$([System.io.path]::DirectorySeparatorChar)$"
```

File: tools/packaging/packaging.psm1 [hunk 2]
```diff
@@ -5062,3 +4765,207 @@ function ConvertTo-PEOperatingSystem {
...
+        $Contents | Out-File -path $logFile -Encoding ascii
+    } else {
+        $logFile = Join-Path -Path $logFolder -ChildPath $newName
+        Copy-Item -Path $Path -Destination $logFile
+    }
+
+    Write-Host "##vso[artifact.upload containerfolder=$newName;artifactname=$newName]$logFile"
+    Write-Verbose "Log file captured as $newName" -Verbose
+}
+class BomRecord {
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:4812]:
```suggestion

# add comment
class BomRecord {
```

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -5062,3 +4765,207 @@ function ConvertTo-PEOperatingSystem {
...
+    if ($noMatch.count -gt 0 -or $needsRemoval.Count -gt 0) {
+        Send-AzdoFile -Path $newBom
+        if ($Fix) {
+            Copy-Item -Path $newBom -Destination $bomFile -Force -Verbose
+        }
+        throw "Please update $bomFile per the above instructions"
+    }
+}
+
+function Test-IsProductFile {
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:4912]: Don't leave this is final PR

File: tools/packaging/boms/windows.json [hunk 4]
```diff
@@ -0,0 +1,3614 @@
...
+  {
+    "Pattern": "Modules/ThreadJob/*.dll",
+    "FileType": "NonProduct"
+  },
+  {
+    "Pattern": "Modules/ThreadJob/*.psd1",
+    "FileType": "NonProduct"
+  },
+  {
+    "Pattern": "mscordaccore_*_*_8.0.23.11008.dll",
```
Comment (TravisEz13) [inline tools/packaging/boms/windows.json:991]:
```suggestion
    "Pattern": "mscordaccore_*.dll",
```

File: tools/packaging/packaging.psm1 [hunk 5]
```diff
@@ -5067,3 +4770,247 @@ function ConvertTo-PEOperatingSystem {
...
+        foreach ($bom in $bomList) {
+            $pattern = $root + [system.io.path]::DirectorySeparatorChar + $bom.GetPattern()
+            Write-Verbose "testing '$filePath -like $pattern"
+            if ($filePath -like $pattern) {
+                $matchingRecord = $bom
+                Write-Verbose "matched '$filePath -like $pattern"
+                $match = $true
+                if ($patternsUsed -notcontains $bom) {
+                    $patternsUsed += $bom
+
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1:4910]: extra space

File: tools/packaging/packaging.psm1 [hunk 6]
```diff
@@ -5067,3 +4770,247 @@ function ConvertTo-PEOperatingSystem {
...
+        Write-Progress -Activity "Testing $BomName BOM" -PercentComplete (100*$currentFileCount/$totalFiles) -Status "Processing $fileName"
+
+        $match = $false
+        [BomRecord] $matchingRecord = $null
+
+        # Test file against each BOM that can still have a match
+        foreach ($bom in $bomList) {
+            $pattern = $root + [system.io.path]::DirectorySeparatorChar + $bom.GetPattern()
+            Write-Verbose "testing '$filePath -like $pattern"
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1:4903]: needs a `-verbose`

**Review by adityapatwardhan — APPROVED** [review]

File: tools/packaging/packaging.psm1 (hunk 5 above)
Comment (TravisEz13) [reply tools/packaging/packaging.psm1:4910]:
```suggestion
```

File: tools/packaging/packaging.psm1 (hunk 6 above)
Comment (TravisEz13) [reply tools/packaging/packaging.psm1:4903]:
```suggestion
```

File: tools/packaging/packaging.psm1 [hunk 7]
```diff
@@ -5067,3 +4770,247 @@ function ConvertTo-PEOperatingSystem {
...
+            if ($filePath -like $pattern) {
+                $matchingRecord = $bom
+                Write-Verbose "matched '$filePath -like $pattern"
```
[the remark this answers is not in the document]
Comment (TravisEz13) [reply tools/packaging/packaging.psm1:4906]:
```suggestion
```

**Pull request merged**